File: case3.jl
function grad!(g::Vector, x::Vector)
    # dummy
end

function optimize(x0, n_iter)
    η = 0.1
    x = copy(x0)
    g = zeros(length(x0))
    for i in 1:n_iter
        grad!(g, x)
        x -= η * g
    end
    x
end

optimize(zeros(2), 3)
x0 = zeros(1000)

File: case3.py
import numpy as np

def set_grad(g, x):
    pass

def optimize(x0, n_iter):
    eta = 0.1
    x = np.copy(x0)
    g = np.zeros(len(x0))
    for i in xrange(n_iter):
        set_grad(g, x)
        x -= eta * g
    return x
